tests/core/test_flow.py

```diff
@@ -2943,6 +2948,19 @@ def test_timeout_actually_stops_execution(executor):
     # the task implementation" we got, but instead do a simple task (create a file) and sleep.
     # This will drastically reduce the brittleness of the test (but not completely).
 
+    # The amount of time to sleep before writing 'invalid' to the file
+    # lower values will decrease test time but increase chances of intermittent failure
+    SLEEP_TIME = 3
+
+    # Determine if the executor is distributed and using daemonic processes which
+    # cannot be cancelled and throw a warning instead. The `__processes` property
+    # is injected in `conftest.py`
+    daemon_process = isinstance(executor, DaskExecutor) and executor.__processes
```

Comment:
This would be equivalent, but avoids the need to modify the executor in the tests.

```suggestion
    daemon_process = (
        isinstance(executor, DaskExecutor)
        and executor.client.scheduler.address.startswith("inproc")
    )
```

Reply:
Ah great that's exactly what I was looking for!